### PR TITLE
[POC] clean peripherals state before call the application

### PR DESCRIPTION
**Pull request by nvlsianpu**
Do some cleanup on peripheral and nvic.

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

### CODE REVIEW

**Comment by nvlsianpu** [comment]
cc @ktaborowski

**Comment by nvlsianpu** [comment]
cc @kl-cruz

File: boot/zephyr/nrf_cleanup.c [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+#if CONFIG_HAS_HW_NRF_UARTE1
+	nrf_uarte_disable(NRF_UARTE1);
+#endif
+}
+
+void nrf_cleanup_nvic(void) {
+	/* Allow any pending interrupts to be recognized */
+	__ISB();
+	__disable_irq();
+	NVIC_Type *nvic = NVIC;
```
Comment (kl-cruz) [inline boot/zephyr/nrf_cleanup.c]: This poiter - ```nvic``` is unnecesary here. You can use ```NVIC``` directly in the code below. Anyway it's just small hint

File: boot/zephyr/nrf_cleanup.c [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+#endif
+#if CONFIG_HAS_HW_NRF_UARTE0
+	nrf_uarte_disable(NRF_UARTE0);
+#endif
+#if CONFIG_HAS_HW_NRF_UARTE1
+	nrf_uarte_disable(NRF_UARTE1);
+#endif
+}
+
+void nrf_cleanup_nvic(void) {
```
Comment (mbolivar) [inline boot/zephyr/nrf_cleanup.c]: This could be done for all Cortex-M, right? Not just nrf.

**Comment by nvlsianpu** [comment]
@hakonfam I'm considering working on the cleanup issue.

File: boot/zephyr/Kconfig [hunk 3]
```diff
@@ -127,6 +127,11 @@ endif #BOOT_SIGNATURE_TYPE_ECDSA_P256
 
 endchoice
 
+config CONFIG_NRF_CLEANUP_PERIPHERAL
```
Comment (ktaborowski) [inline boot/zephyr/Kconfig]: Shouldn't it be named NRF_CLEANUP_PERIPHERAL (w/o CONFIG_)?

File: boot/zephyr/nrf_cleanup.c [hunk 4]
```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <hal/nrf_rtc.h>
+
```
Comment (ktaborowski) [inline boot/zephyr/nrf_cleanup.c]: I think hal/uart header is also needed here

File: boot/zephyr/nrf_cleanup.c (hunk 2 above)
Comment (nvlsianpu) [reply boot/zephyr/nrf_cleanup.c]: https://github.com/JuulLabs-OSS/mcuboot/pull/683

**Pull request closed without merging**